src/common/libraries/MessageLib.sol

```diff
@@ -1191,15 +1197,96 @@ library MessageLib {
+
+    struct TriggerSubmitQueuedShares {
+        uint64 poolId;
+        bytes16 scId;
+    }
+
+    function deserializeTriggerSubmitQueuedShares(bytes memory data)
+        internal
+        pure
+        returns (TriggerSubmitQueuedShares memory)
+    {
+        require(messageType(data) == MessageType.TriggerSubmitQueuedShares, UnknownMessageType());
+        return TriggerSubmitQueuedShares({poolId: data.toUint64(1), scId: data.toBytes16(9)});
+    }
+
+    function serialize(TriggerSubmitQueuedShares memory t) internal pure returns (bytes memory) {
+        return abi.encodePacked(MessageType.TriggerSubmitQueuedShares, t.poolId, t.scId);
+    }
+
+    //---------------------------------------
+    //    TriggerSubmitQueuedAssets
+    //---------------------------------------
+
+    struct TriggerSubmitQueuedAssets {
+        uint64 poolId;
+        bytes16 scId;
+        uint128 assetId;
+    }
+
+    function deserializeTriggerSubmitQueuedAssets(bytes memory data)
+        internal
+        pure
+        returns (TriggerSubmitQueuedAssets memory)
+    {
+        require(messageType(data) == MessageType.TriggerSubmitQueuedAssets, UnknownMessageType());
+        return
+            TriggerSubmitQueuedAssets({poolId: data.toUint64(1), scId: data.toBytes16(9), assetId: data.toUint128(25)});
+    }
+
+    function serialize(TriggerSubmitQueuedAssets memory t) internal pure returns (bytes memory) {
+        return abi.encodePacked(MessageType.TriggerSubmitQueuedAssets, t.poolId, t.scId, t.assetId);
+    }
+
+    //---------------------------------------
+    //    SetSharesQueue
+    //---------------------------------------
+
+    struct SetSharesQueue {
+        uint64 poolId;
+        bytes16 scId;
+        bool enabled;
+    }
+
+    function deserializeSetSharesQueue(bytes memory data) internal pure returns (SetSharesQueue memory) {
+        require(messageType(data) == MessageType.SetSharesQueue, UnknownMessageType());
+        return SetSharesQueue({poolId: data.toUint64(1), scId: data.toBytes16(9), enabled: data.toBool(25)});
+    }
+
+    function serialize(SetSharesQueue memory t) internal pure returns (bytes memory) {
+        return abi.encodePacked(MessageType.SetSharesQueue, t.poolId, t.scId, t.enabled);
+    }
+
+    //---------------------------------------
+    //    SetAssetsQueue
+    //---------------------------------------
+
+    struct SetAssetsQueue {
+        uint64 poolId;
+        bytes16 scId;
+        bool enabled;
+    }
+
+    function deserializeSetAssetsQueue(bytes memory data) internal pure returns (SetAssetsQueue memory) {
+        require(messageType(data) == MessageType.SetAssetsQueue, UnknownMessageType());
+        return SetAssetsQueue({poolId: data.toUint64(1), scId: data.toBytes16(9), enabled: data.toBool(25)});
+    }
+
+    function serialize(SetAssetsQueue memory t) internal pure returns (bytes memory) {
+        return abi.encodePacked(MessageType.SetAssetsQueue, t.poolId, t.scId, t.enabled);
```

Comment:
CR 

@onnovisser Sorry for the late change. Can we merge the two into one and only enable queue globally? Just came up in our sync!